src/components/OverviewCard.jsx

```diff
@@ -11,9 +23,131 @@ import {
   retrieveResourceFileMetadata,
 } from '../utils';
 
+// axios settings
+axios.defaults.withCredentials = true
+
 const OverviewCard = ({
   date, category, settingsToggle, itemIndex, siteName, fileName, isResource, isHomepage
 }) => {
+  const dropdownRef = useRef(null)
+  const fileMoveDropdownRef = useRef(null)
+  const categoryInputRef = useRef(null)
+  const [allCategories, setAllCategories] = useState()
+  const [newCategory, setNewCategory] = useState('')
+  const [errorMessage, setErrorMessage] = useState('')
+  const [canShowDropdown, setCanShowDropdown] = useState(false)
+  const [canShowFileMoveDropdown, setCanShowFileMoveDropdown] = useState(false)
+  const [canShowDeleteWarningModal, setCanShowDeleteWarningModal] = useState(false)
+  const baseApiUrl = `${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}${category ? isResource ? `/resources/${category}` : `/collections/${category}` : ''}`
+
+  useEffect(() => {
+    const fetchData = async () => {
+      // Retrieve the list of all page/resource categories for use in the dropdown options.
+      if (isResource) {
+        const resourcesResp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/resources`);
+        const { resources: allCategories } = resourcesResp.data;
+        setAllCategories(allCategories.map((category) => category.dirName))
+      } else {
+        const collectionsResp = await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}/collections`);
+        const { collections: collectionCategories } = collectionsResp.data;
+        setAllCategories(collectionCategories)
+      }
+    }
+    fetchData()
+  }, [])
+
+  useEffect(() => {
+    if (canShowFileMoveDropdown) fileMoveDropdownRef.current.focus()
+    if (canShowDropdown) dropdownRef.current.focus()
+  }, [canShowFileMoveDropdown, canShowDropdown])
+
+  const moveFile = async (chosenCategory) => {
+    try {
+      // Retrieve data from existing page/resource
+      const resp = await axios.get(`${baseApiUrl}/pages/${fileName}`);
+
+      const { content, sha } = resp.data;
+      const base64DecodedContent = Base64.decode(content);
```

Comment:
I'm getting this error when I click on `Edit Details` in the dropdown:

```
 DOMException: Failed to execute 'atob' on 'Window': The string to be decoded is not correctly encoded.
    at _atob (http://localhost:3000/static/js/0.chunk.js:55677:19)
    at _decode (http://localhost:3000/static/js/0.chunk.js:55691:17)
    at Object.decode (http://localhost:3000/static/js/0.chunk.js:55695:12)
    at ComponentSettingsModal.componentDidMount (http://localhost:3000/static/js/main.chunk.js:1212:87)
```

Not sure if it's related to this highlighted line of code - have you encountered this problem before?

Reply:
This was a difficult bug to reproduce consistently - turns out this only occurs when the icon of Edit Details was being clicked, because I forgot to set the id of the icon as well, causing the wrong backend endpoint to be called.

Resolved in https://github.com/isomerpages/isomercms-frontend/pull/219/commits/efa66d860347fcac78a6468b6dacab8c4f0b61e7.